data/models/declaration.py

```diff
@@ -0,0 +1,235 @@
+        REJECTED = "REJECTED", "Rejeté"
+        APPROVED = "APPROVED", "Validé"
+
+    class RejectionReason(models.TextChoices):
+        MISSING_DATA = "MISSING_DATA", "Le dossier manque des données nécessaires"
+        MEDICINE = "MEDICINE", "Le complément répond à la définition du médicament"
+        INCOMPATIBLE_RECOMMENDATIONS = "INCOMPATIBLE_RECOMMENDATIONS", "Recommandations d'emploi incompatibles"
+
+    status = models.CharField(
+        max_length=50,
+        choices=DiagnosticStatus.choices,
+        default=DiagnosticStatus.DRAFT,
+        verbose_name="Status",
+    )
+    author = models.ForeignKey(
+        get_user_model(), null=True, on_delete=models.SET_NULL, verbose_name="auteur", related_name="declarations"
+    )
+    company = models.ForeignKey(
+        Company, null=True, on_delete=models.SET_NULL, verbose_name="entreprise", related_name="declarations"
+    )
+
+    # Champs adresse :
+    # On ne prend pas la même classe que l'adresse de la compagnie car on a besoin
+    # d'avoir ces champs facultatifs
+    address = models.TextField("adresse", blank=True, help_text="numéro et voie")
+    additional_details = models.TextField(
+        "complément d’adresse",
+        blank=True,
+        help_text="bâtiment, immeuble, escalier et numéro d’appartement",
+    )
+    postal_code = models.TextField("code postal", blank=True, max_length=10)
+    city = models.TextField("ville ou commune", blank=True)
+    cedex = models.TextField("CEDEX", blank=True)
+    country = models.TextField("pays", blank=True, choices=CountryChoices)
+
+    name = models.TextField(blank=True, verbose_name="nom du produit")
+    brand = models.TextField(blank=True, verbose_name="marque")
+    gamme = models.TextField(blank=True, verbose_name="gamme")
+    flavor = models.TextField(blank=True, verbose_name="arôme")
+    description = models.TextField(blank=True, verbose_name="description")
+    #########
+
+    galenic_formulation = models.TextField(
+        blank=True, verbose_name="forme galénique"
+    )  # TODO : à terme mettre des valeurs de la DB
+    unit_quantity = models.FloatField(null=True, verbose_name="poids ou volume d'une unité de consommation")
+    unit_measurement = models.ForeignKey(
+        SubstanceUnit,
+        null=True,
+        on_delete=models.RESTRICT,
+        verbose_name="unité de mesure pour une unité de consommation",
+    )
+
+    conditioning = models.TextField(blank=True, verbose_name="conditionnements")
+    daily_recommended_dose = models.TextField(blank=True, verbose_name="dose journalière recommandée")
+    minimum_duration = models.TextField(blank=True, verbose_name="durabilité minimale / DLUO (en mois)")
+
+    instructions = models.TextField(blank=True, verbose_name="mode d'emploi")
+    warning = models.TextField(blank=True, verbose_name="mise en garde et avertissement")
+
+    populations = models.ManyToManyField(Population, blank=True, verbose_name="populations cible")
```

Comment:
Pour tous les relatedField, pour rester DRY, on peut utiliser le verbose_name de la classe :
`populations = models.ManyToManyField(Population, blank=True, verbose_name=Population._meta.verbose_name_plural)`(en ayant défini le verbose_name_plural dans la class `population`) bien sûr.

Evidemment, parfois c'est pas pertinent parce qu'on veut spécifier un nom différent.

Reply:
Ah oui bonne idée pour quand le `verbose_name` soit le même côté modèle. Pour les many-to-many j'aimerais rester sur le label un peu plus descriptif. Par contre pour certains `ForeignKey` des modèles déclarés je veux bien mettre le verbose_name : https://github.com/betagouv/complements-alimentaires/pull/364/commits/a31311dca5e95d77db35a8f122444bf3d42b2f12

Reply:
(et pour info, je sais pas comment  faire cette technique avec le modèle `User` qu'on veut surtout pas importer comme ça pour éviter les imports circulaires relous ^^)